### PR TITLE
AG-1173 added TEP logic to gene hero

### PR DESCRIPTION
**Pull request by sagely1**

### CODE REVIEW

File: src/app/features/genes/components/gene-hero/gene-hero.component.spec.ts [hunk 1]
```diff
@@ -28,12 +29,33 @@ describe('Component: Gene Hero', () => {
...
+
+  it('should only add TEP text if nominations is null and either is_tep or is_adi is true', () => {
+    component.gene = geneMock3;
+    fixture.detectChanges();
+    
+    const el = element.querySelector('.gene-hero-nominated') as HTMLElement;
+    
+    const expected = 'Selected for Target Enabling Resource Development';
+    expect(el.textContent).toBe(expected);
+  });
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-hero/gene-hero.component.spec.ts]: Nice tests! You could also consider adding tests to show that 1) only "Nominated Target" is shown when is_tep and is_adi are both false, and 2) nothing is displayed when nominations is null and both is_tep and is_adi are false. Up to you though!

File: src/app/testing/gene-mocks.ts [hunk 2]
```diff
@@ -3144,6 +3146,137 @@ export const geneMock2: Gene = {
     },
   ],
   nominations: 4,
+  is_adi: false,
+  is_tep: true
+};
+
+export const geneMock3: Gene = {
```
Comment (hallieswan) [inline src/app/testing/gene-mocks.ts:3153]: From [the build log](https://app.travis-ci.com/github/Sage-Bionetworks/Agora/builds/265394612), it looks like this mock might be missing `nominatedtarget` and `nominations` fields:

```log
ERROR in src/app/testing/gene-mocks.ts:3153:14 - error TS2739: Type '{ _id: string; ensembl_gene_id: string; name: string; 
summary: string; hgnc_symbol: string; alias: string[]; isIGAP: false; haseqtl: true; isAnyRNAChangedInADBrain: true; 
rna_brain_change_studied: true; ... 5 more ...; is_tep: true; }' is missing the following properties from type 'Gene':
nominatedtarget, nominations

3153 export const geneMock3: Gene = {
                  ~~~~~~~~~
```

**Review by hallieswan — COMMENTED** [review]
Looks good! Just looks like the new mock may need to be tweaked

File: src/app/testing/gene-mocks.ts (hunk 2 above)
Comment (sagely1) [reply src/app/testing/gene-mocks.ts:3153]: Looks like I didn't commit all my changes.  Good catch!

File: src/app/features/genes/components/gene-hero/gene-hero.component.spec.ts (hunk 1 above)
Comment (sagely1) [reply src/app/features/genes/components/gene-hero/gene-hero.component.spec.ts]: My latest commit should have these missing tests.  thanks!

**Review by hallieswan — APPROVED** [review]
Looks good! Could also add a test to check when `nominations` is null and both `is_adi` and `is_tep` are false, but up to you!

**Comment by sagely1** [comment]
> Looks good! Could also add a test to check when `nominations` is null and both `is_adi` and `is_tep` are false, but up to you!

Added border condition unit test as per @hallieswan
